src/measurement/sev_hashes.rs

```diff
@@ -0,0 +1,214 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Operations to handle OVMF SEV-HASHES
+use openssl::sha::sha256;
+use serde::Serialize;
+use std::fs::File;
+use std::{
+    convert::{TryFrom, TryInto},
+    io::Read,
+    mem::size_of,
+    path::PathBuf,
+    str::FromStr,
+};
+
+use uuid::{uuid, Uuid};
+
+use crate::error::*;
+
+type Sha256Hash = [u8; 32];
+
+/// GUID stored as little endian
+#[derive(Debug, Clone, Copy, Serialize, Default)]
+struct GuidLe {
+    _data: [u8; 16],
+}
+
+impl TryFrom<&Uuid> for GuidLe {
+    type Error = MeasurementError;
+
+    fn try_from(value: &Uuid) -> Result<Self, Self::Error> {
+        let guid = value.to_bytes_le();
+        let guid = guid.as_slice();
+        Ok(Self {
+            _data: guid.try_into()?,
+        })
+    }
+}
+
+impl FromStr for GuidLe {
+    type Err = MeasurementError;
+
+    fn from_str(guid: &str) -> Result<Self, MeasurementError> {
+        let guid = Uuid::try_from(guid)?;
+        let guid = guid.to_bytes_le();
+        let guid = guid.as_slice();
+        Ok(Self {
+            _data: guid.try_into()?,
+        })
+    }
+}
+
+/// SEV hash table entry
+#[repr(C)]
+#[derive(Debug, Clone, Copy, Serialize, Default)]
+struct SevHashTableEntry {
+    /// Entry GUID
+    guid: GuidLe,
+    /// Length
+    length: u16,
+    /// Hash
+    hash: Sha256Hash,
+}
+
+impl SevHashTableEntry {
+    fn new(guid: &Uuid, hash: Sha256Hash) -> Result<Self, MeasurementError> {
+        Ok(Self {
+            guid: GuidLe::try_from(guid)?,
+            length: std::mem::size_of::<SevHashTableEntry>() as u16,
+            hash,
+        })
+    }
+}
+
+/// Table of SEV hashes
+#[repr(C)]
+#[derive(Debug, Clone, Copy, Serialize, Default)]
+struct SevHashTable {
+    /// GUID
+    guid: GuidLe,
+    /// Length
+    length: u16,
+    /// Cmd line append table entry
+    cmdline: SevHashTableEntry,
+    /// initrd table entry
+    initrd: SevHashTableEntry,
+    /// Kernel table entry
+    kernel: SevHashTableEntry,
+}
```

Comment:
```suggestion
/// Table of SEV hashes
#[repr(C)]
#[derive(Debug, Clone, Copy, Serialize, Default)]
struct SevHashTable {
    /// GUID of ...
    guid: GuidLe,
    /// Length of ...
    length: u16,
    /// Cmd line append table entry
    cmdline: SevHashTableEntry,
    /// initrd table entry
    initrd: SevHashTableEntry,
    /// Kernel table entry
    kernel: SevHashTableEntry,
}
```

Likewise, a bit more description of the struct members.